From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: reexport vsock_pending_work
Patch-mainline: never, kabi
References: kabi

In netfilter-stable-18_08_17, commit
455f05ecd2b219e9a216050796d30c830d9bc393 (vsock: split dwork to avoid
reinitializations) removed export of vsock_pending_work.

Reexport the function in case somebody else uses that.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/af_vsock.h   |    1 +
 net/vmw_vsock/af_vsock.c |    3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/include/net/af_vsock.h
+++ b/include/net/af_vsock.h
@@ -82,6 +82,7 @@ struct vsock_sock {
 
 s64 vsock_stream_has_data(struct vsock_sock *vsk);
 s64 vsock_stream_has_space(struct vsock_sock *vsk);
+void vsock_pending_work(struct work_struct *work);
 struct sock *__vsock_create(struct net *net,
 			    struct socket *sock,
 			    struct sock *parent,
--- a/net/vmw_vsock/af_vsock.c
+++ b/net/vmw_vsock/af_vsock.c
@@ -449,7 +449,7 @@ static int vsock_send_shutdown(struct so
 	return transport->shutdown(vsock_sk(sk), mode);
 }
 
-static void vsock_pending_work(struct work_struct *work)
+void vsock_pending_work(struct work_struct *work)
 {
 	struct sock *sk;
 	struct sock *listener;
@@ -496,6 +496,7 @@ out:
 	sock_put(sk);
 	sock_put(listener);
 }
+EXPORT_SYMBOL_GPL(vsock_pending_work);
 
 /**** SOCKET OPERATIONS ****/
 
